gcsfs/core.py

```diff
@@ -1410,6 +1398,9 @@ async def _find(
         else:
             _prefix = key
 
+        if _prefix != "" and await self._isdir(f"{bucket}/{_prefix}"):
```

Comment:
Is there no way to avoid isdir here? This doubles the number of calls for a simple list - and isdir surely is doing a list op anyway!

Reply:
You're right, it's not useful to double the calls there. That's why I initially indicated that a better solution should probably exist ;)

I updated the code with [this commit](https://github.com/fsspec/gcsfs/pull/576/commits/d1ff25f9870a5a09f287a68bd52183a699c41dfd) to try out another solution inspured by [the `_find` method in s3fs](https://github.com/fsspec/s3fs/blob/main/s3fs/core.py#L818:L823) which first fetch the path as if it is a directory then calls the `_info` method on the path to check if it's a file and return it accordingly. In gcsfs, we can't call `_info` because some tests are expecting versioned files (`_info` does not support versioned files) or a prefixed search. So I simplify moved the initial implementation into a block executed if the first search (when we treat path as a directory) does not return anything.

Let me know if that works for you. This commit is tested against [this PR](https://github.com/fsspec/filesystem_spec/pull/1340) to fix [this test](https://github.com/fsspec/filesystem_spec/actions/runs/5973337483/job/16205407770?pr=1340#step:6:484).